Linker errors in example problems when using static MUMPS

Linking with a parallel MUMPS build from source results in unresolved symbols, since MUMPS is placed after BLAS and LAPACK in the linker invocation:

```
[56/102] Linking CXX executable examples/stokes
FAILED: examples/stokes
: && /usr/bin/c++ -fdiagnostics-color=always -O3 -DNDEBUG -pthread examples/CMakeFiles/ads-example-stokes.dir/stokes/main.cpp.o -o examples/stokes  -Wl,-rpath,/usr/lib/x86_64-linux-gnu/openmpi/lib  libads.a  /usr/lib/x86_64-linux-gnu/liblapack.so  /usr/lib/x86_64-linux-gnu/libblas.so  /opt/fmt-7.1.3/lib/libfmt.a  /opt/MUMPS-5.4.1-par/lib/libdmumps.a  /opt/MUMPS-5.4.1-par/lib/libmumps_common.a  /usr/lib/x86_64-linux-gnu/libmetis.so  /usr/lib/libparmetis.so  /usr/lib/x86_64-linux-gnu/libptesmumps.so  /usr/lib/x86_64-linux-gnu/libscotch.so  /usr/lib/x86_64-linux-gnu/libptscotch.so  /usr/lib/x86_64-linux-gnu/libptscotcherr.so  /opt/MUMPS-5.4.1-par/lib/libpord.a  -lgfortran  -lm  -lgcc_s  -lgcc  -lquadmath  -lc  -lm  -lgcc_s  -lgcc  -lquadmath  -lc  /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_cxx.so  /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_usempif08.so  /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_usempi_ignore_tkr.so  /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_mpifh.so  /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so  /usr/lib/gcc/x86_64-linux-gnu/9/libgomp.so  /usr/lib/x86_64-linux-gnu/libpthread.so  /opt/Galois-6.0/lib/libgalois_shmem.a  -lrt  -lpthread  /usr/lib/x86_64-linux-gnu/libnuma.so && :
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/liblapack.so: undefined reference to symbol 'idamax_'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libblas.so: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
[63/102] Building CXX object examples/CMakeFiles/ads-example-dg_laplace.dir/dg/laplace.cpp.o
ninja: build stopped: subcommand failed.
```

The sequential version suffers the same issue, and also lacks the `mpiseq` library, which is an MPI, BLACS, and ScaLAPACK stub. Handling sequential MUMPS correctly requires deeper changes since it involves MPI. This should be cleaned up while refactoring MPI usage in `mumps.hpp`.

--- a/cmake/Modules/FindMUMPS.cmake
+++ b/cmake/Modules/FindMUMPS.cmake
@@ -16,7 +16,7 @@ find_package_handle_standard_args(MUMPS
 
 set(MUMPS_LIBRARIES ${MUMPS_LIBRARY})
 
-# mumps_common may be a separate library
+# mumps_common is a separate library
 find_library(MUMPS_COMMON_LIBRARY NAMES mumps_common)
 
 if (MUMPS_COMMON_LIBRARY)
@@ -24,14 +24,24 @@ if (MUMPS_COMMON_LIBRARY)
 endif()
 
 # Find scalapack
-find_library(MUMPS_SCALAPACK NAMES scalapack)
+find_library(MUMPS_SCALAPACK NAMES scalapack scalapack-openmpi)
 
 if (MUMPS_SCALAPACK)
   list(APPEND MUMPS_LIBRARIES ${MUMPS_SCALAPACK})
 endif()
 
+# Find BLAS and LAPACK
+find_package(BLAS REQUIRED)
+find_package(LAPACK REQUIRED)
+
+list(APPEND MUMPS_LIBRARIES ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
+
 # Try to find ordering libraries
-set(ORDERING_LIBS metis parmetis ptesmumps scotch ptscotch ptscotcherr pord)
+set(ORDERING_LIBS
+  pord
+  metis scotch scotcherr esmumps
+  parmetis ptscotch ptscotcherr ptesmumps
+)
 
 foreach (lib IN LISTS ORDERING_LIBS)
   find_library(MUMPS_${lib}_LIBRARY NAMES ${lib})

